src/box/lua/trigger.c

```diff
@@ -0,0 +1,242 @@
+ * If an event name is passed, a table contains only one key which is passed
+ * argument, if there is an event with such a name, or returned table is empty,
+ * if the event does not exist.
+ */
+static int
+luaT_trigger_info(struct lua_State *L)
+{
+	if (lua_gettop(L) > 1)
+		luaL_error(L, "expected 0 or 1 arguments");
+	if (lua_gettop(L) == 0) {
+		lua_createtable(L, 0, 0);
+		bool ok = event_foreach(trigger_info_push_event, L);
+		assert(ok);
+		(void)ok;
+	} else {
+		const char *event_name = luaL_checkstring(L, 1);
+		struct event *event = event_get(event_name, false);
+		if (event == NULL || !event_has_triggers(event)) {
+			lua_createtable(L, 0, 0);
+			return 1;
+		}
+		lua_createtable(L, 0, 1);
+		trigger_info_push_event(event, L);
+	}
+	return 1;
+}
+
+static const char *event_trigger_iterator_typename =
+	"trigger.event_trigger_iterator";
+
+/**
+ * Gets lua_event_trigger_iterator from Lua stack with type check.
+ */
+static inline struct event_trigger_iterator *
+lua_check_event_trigger_iterator(struct lua_State *L, int idx)
+{
+	return luaL_checkudata(L, idx, event_trigger_iterator_typename);
+}
+
+/**
+ * Takes an iterator step.
+ */
+static int
+luaT_trigger_iterator_next(struct lua_State *L)
+{
+	struct event_trigger_iterator *it =
+		lua_check_event_trigger_iterator(L, 1);
+	struct func_adapter *trigger = NULL;
+	const char *name = NULL;
+	if (event_trigger_iterator_next(it, &trigger, &name)) {
+		lua_pushstring(L, name);
+		func_adapter_lua_get_func(trigger, L);
+		return 2;
+	}
+	return 0;
+}
+
+/**
+ * Destroys an iterator.
+ */
+static int
+luaT_trigger_iterator_gc(struct lua_State *L)
+{
+	struct event_trigger_iterator *it =
+		lua_check_event_trigger_iterator(L, 1);
+	event_trigger_iterator_destroy(it);
+	TRASH(it);
+	return 0;
+}
+
+/**
+ * Creates iterator over triggers of event with passed name.
+ * The iterator yields a pair [trigger_name, trigger_handler].
+ * Return next method of iterator and iterator itself.
+ */
+static int
+luaT_trigger_pairs(struct lua_State *L)
+{
+	if (lua_gettop(L) != 1)
+		luaL_error(L, "expected 1 argument");
+	const char *event_name = luaL_checkstring(L, 1);
+	struct event *event = event_get(event_name, false);
+	if (event == NULL)
+		return 0;
+	lua_pushcfunction(L, luaT_trigger_iterator_next);
+	struct event_trigger_iterator *it = lua_newuserdata(L, sizeof(*it));
+	event_trigger_iterator_create(it, event);
+	luaL_getmetatable(L, event_trigger_iterator_typename);
+	lua_setmetatable(L, -2);
+	return 2;
+}
+
+void
+box_lua_trigger_init(struct lua_State *L)
+{
+	const struct luaL_Reg module_funcs[] = {
+		{"set", luaT_trigger_set},
+		{"del", luaT_trigger_del},
+		{"call", luaT_trigger_call},
+		{"info", luaT_trigger_info},
+		{"pairs", luaT_trigger_pairs},
+		{NULL, NULL}
+	};
+	luaT_newmodule(L, "trigger", module_funcs);
+	lua_pop(L, 1);
+	const struct luaL_Reg trigger_iterator_methods[] = {
+		{"__gc", luaT_trigger_iterator_gc},
+		{"next", luaT_trigger_iterator_next},
```

Comment:
`next` is never used. Let's remove this export.

Reply:
Done.